tests/app/dao/notification_dao/test_notification_dao.py

```diff
@@ -2034,39 +2039,42 @@ def test_update_notification_status_by_id_cannot_update_status_out_of_order_with
     ],
 )
 def test_update_notification_status_by_id_can_update_status_in_order_when_given_valid_values(
-    current_status,
-    next_status,
     sample_notification,
     sample_template,
+    current_status,
+    next_status,
+    use_current_status,
 ):
-    reference_tuple = (str(uuid4()), str(uuid4()))
-    notification_list = []
-
-    for ref in reference_tuple:
-        # create first notification object
-        sample_notification(template=sample_template(), reference=ref, sent_at=datetime.now(), status=current_status)
-
-        # add the notification object to the notification list
-        notification_list.append(dao_get_notification_by_reference(ref))
-
-        # make sure we pass all the init setup
-        i = len(notification_list) - 1
-        assert isinstance(notification_list[i], Notification)
-        assert notification_list[i].status == current_status
-
-    # attempt update without the condition current state
-    update_notification_status_by_id(notification_id=notification_list[0].id, status=next_status)
+    reference = str(uuid4())
+    initial_status_reason = '' if (current_status == NOTIFICATION_DELIVERED) else 'Because I said so!'
+    final_status_reason = initial_status_reason if (next_status == current_status) else 'just because'
 
-    # attempt update with the conditional current state
-    update_notification_status_by_id(
-        notification_id=notification_list[1].id, status=next_status, current_status=current_status
+    notification = sample_notification(
+        template=sample_template(),
+        reference=reference,
+        sent_at=datetime.now(),
+        status=current_status,
+        status_reason=initial_status_reason,
     )
+    assert notification.status == current_status
+    assert notification.status_reason == initial_status_reason
+
+    if use_current_status:
+        update_notification_status_by_id(
+            notification_id=notification.id,
+            status=next_status,
+            status_reason=final_status_reason,
+            current_status=current_status,
+        )
+    else:
+        update_notification_status_by_id(
+            notification_id=notification.id, status=next_status, status_reason=final_status_reason
+        )
 
-    # get the notification object and make sure the values have changed
-    for ref in reference_tuple:
-        notification = dao_get_notification_by_reference(ref)
-        assert isinstance(notification, Notification)
-        assert notification.status == next_status
```

Comment:
Same

Reply:
This is no longer appropriate because I added a new parametrization.